core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -78,6 +81,13 @@ public void accept(String file) {
   RemoveSnapshots(TableOperations ops) {
     this.ops = ops;
     this.base = ops.current();
+
+    ValidationException.check(
+        PropertyUtil.propertyAsBoolean(base.properties(), GC_ENABLED, GC_ENABLED_DEFAULT),
+        "Not allowed to expire snapshots as the garbage collection is disabled for this table. " +
+        "Make sure this table is the exclusive owner of its data files before allowing garbage collection. " +
+        "If the table was created using the SNAPSHOT operation, it is not safe to expire snapshots in it since " +
+        "this may remove files in the original table.");
```

Comment:
I think it's still ok to put in instructions for disabling this parameter as a workaround. Just helps a user to know how to unblock themselves just in case. Also nit: "Cannot expire snapshots" would follow our pattern for errors in other places

Reply:
Hmm... I'm struggling a bit with someone being allowed to update this in certain cases. It seems like a very sharp edge to expose/suggest.

Reply:
I think it is reasonable to allow people to update it. Having properties that can only be set by the library itself seems like we're asking for people to go edit files by hand when they have a need. That said, I think it would be fine to not suggest that this can be changed. If they want to track it down by going to the code and reading this check, then fine. But we can change this to "Cannot expire snapshots: GC is disabled (deleting files may corrupt other tables)"

Reply:
I agree that suggesting is dangerous but letting do this should be fine too.